spock-core/src/main/java/org/spockframework/mock/runtime/ProxyBasedMockFactory.java

```diff
@@ -45,12 +58,15 @@ public Object create(Class<?> mockType, List<Class<?>> additionalInterfaces, @Nu
 
     if (mockType.isInterface()) {
       proxy = createDynamicProxyMock(mockType, additionalInterfaces, constructorArgs, mockInterceptor, classLoader);
+    } else if (byteBuddyAvailable) {
+      proxy = ByteBuddyMockFactory.createMock(mockType, additionalInterfaces,
+        constructorArgs, mockInterceptor, classLoader, useObjenesis);
     } else if (cglibAvailable) {
       proxy = CglibMockFactory.createMock(mockType, additionalInterfaces,
           constructorArgs, mockInterceptor, classLoader, useObjenesis);
     } else {
       throw new CannotCreateMockException(mockType,
-          ". Mocking of non-interface types requires the CGLIB library. Please put cglib-nodep-2.2 or higher on the class path."
+          ". Mocking of non-interface types requires the CGLIB library. Please put Byte Buddy (at least 1.4.0) or cglib-nodep-2.2 or higher on the class path."
```

Comment:
Not really your issue but if you are already touching it. The message should say that we need at least version 3.2 of cglib.